packages/plugins/src/foundry/utils/index.ts

```diff
@@ -1171,28 +1187,30 @@ export const assertValidAccountAccesses = (
  * through the Deterministic Deployment Proxy (i.e. Foundry's default `CREATE2` deployer).
  */
 export const isCreate2AccountAccess = (
-  accountAccess: AccountAccess,
-  nextAccountAccess?: AccountAccess
+  root: AccountAccess,
+  nested: Array<AccountAccess>
 ): boolean => {
+  if (nested.length === 0) {
+    return false
+  }
+  const nextAccountAccess = nested[0]
+
   // The first `AccountAccess` must be a `Call` to the Deterministic Deployment Proxy, and must have
   // data that's at least 32 bytes long, since the data is a 32-byte salt appended with the
   // contract's init code.
   const isCreate2Call =
-    accountAccess.kind === AccountAccessKind.Call &&
-    accountAccess.account === DETERMINISTIC_DEPLOYMENT_PROXY_ADDRESS &&
-    getBytesLength(accountAccess.data) >= 32
+    root.kind === AccountAccessKind.Call &&
+    root.account === DETERMINISTIC_DEPLOYMENT_PROXY_ADDRESS &&
+    getBytesLength(root.data) >= 32
   if (!isCreate2Call) {
     return false
   }
 
-  const { create2Address } = decodeDeterministicDeploymentProxyData(
-    accountAccess.data
-  )
+  const { create2Address } = decodeDeterministicDeploymentProxyData(root.data)
 
   // The next `AccountAccess` must be a `Create` kind, and must be sent from the Deterministic
   // Deployment Proxy to the expected `CREATE2` address.
   return (
-    nextAccountAccess !== undefined &&
```

Comment:
not necessary anymore b/c `nextAccountAccess` was previously possibly undefined, which isn't the case anymore